test/symbolics.jl

```diff
@@ -0,0 +1,131 @@
+    @test typeof(@inferred(measurement(0, x_err))) == Measurement{Num}
+    @test typeof(@inferred(measurement(0.0, x_err))) == Measurement{Num}
+    @test typeof(@inferred(measurement(big(0), x_err))) == Measurement{Num}
+
+    @test iszero(@inferred(measurement(x)).err)
+    @test @inferred(measurement(x)).tag === UInt64(0)
+    @test length(@inferred(measurement(x)).der) == 0
+    @test !(@inferred(measurement(x, x_err)).tag === UInt64(0))
+    @test length(@inferred(measurement(x, 0)).der) == 0
+    @test length(@inferred(measurement(x, x_err)).der) == 1
+end
+
+@testset "Type representation" begin
+    @variables x_val, x_err
+
+    x = measurement(x_val, x_err)
+
+    # test pretty printing at the REPL
+    @test_throws ErrorException repr(x, context=:error_digits=>-4)
+    @test repr(x) == "x_val ± x_err"
+    @test repr(x, context=:compact=>true) == "x_val±x_err"
+    for error_digits in (0, 7)
+        @test repr(x, context=:error_digits=>error_digits) == "x_val ± x_err"
+        @test repr(x, context=IOContext(stdout, :error_digits=>error_digits, :compact=>true)) == "x_val±x_err"
+    end
+
+    @test repr("text/plain", x) == "x_val ± x_err"
+    @test repr("text/plain", x, context=:compact=>true) == "x_val±x_err"
+    for error_digits in (0, 7)
+        @test repr("text/plain", x, context=:error_digits=>error_digits) == "x_val ± x_err"
+        @test repr("text/plain", x, context=IOContext(stdout, :error_digits=>error_digits, :compact=>true)) == "x_val±x_err"
+    end
+
+    @test repr("text/latex", x) == "\$x_val \\pm x_err\$"
+    @test repr("text/latex", x, context=:compact=>true) == "\$x_val\\pmx_err\$"
+    for error_digits in (0, 7)
+        @test repr("text/latex", x, context=:error_digits=>error_digits) == "\$x_val \\pm x_err\$"
+        @test repr("text/latex", x, context=IOContext(stdout, :error_digits=>error_digits, :compact=>true)) == "\$x_val\\pmx_err\$"
+    end
+
+    for mime in ("text/x-tex", "text/x-latex")
+        @test repr(mime, x) == "x_val \\pm x_err"
+        @test repr(mime, x, context=:compact=>true) == "x_val\\pmx_err"
+        for error_digits in (0, 7)
+          @test repr(mime, x, context=:error_digits=>error_digits) == "x_val \\pm x_err"
+          @test repr(mime, x, context=IOContext(stdout, :error_digits=>error_digits, :compact=>true)) == "x_val\\pmx_err"
+        end
+    end
+
+    @variables y_val, y_err
+    y = measurement(y_val, y_err)
+    c = complex(x, y)
+
+    @test_throws ErrorException repr(c, context=:error_digits=>-4)
+    @test repr(c) == "(x_val ± x_err) + (y_val ± y_err)im"
+    @test repr(c, context=:compact=>true) == "(x_val±x_err)+(y_val±y_err)im"
+    for error_digits in (0, 7)
+        @test repr(c, context=:error_digits=>error_digits) == "(x_val ± x_err) + (y_val ± y_err)im"
+        @test repr(c, context=IOContext(stdout, :error_digits=>error_digits, :compact=>true)) == "(x_val±x_err)+(y_val±y_err)im"
+    end
```

Comment:
This is the test coverage for https://github.com/JuliaPhysics/Measurements.jl/pull/144/files#r1208667783
```
Type representation: Error During Test at /home/lebedevri/.julia/packages/Measurements/BZpJb/test/symbolics.jl:70
  Test threw exception
  Expression: repr(c) == "(x_val ± x_err) + (y_val ± y_err)im"
  TypeError: non-boolean (Num) used in boolean context
  Stacktrace:
   [1] macro expansion
     @ ~/.julia/packages/Measurements/BZpJb/test/runtests.jl:1066 [inlined]
   [2] macro expansion
     @ /builddirs/julia/usr/share/julia/stdlib/v1.9/Test/src/Test.jl:1498 [inlined]
   [3] top-level scope
     @ ~/.julia/packages/Measurements/BZpJb/test/runtests.jl:1066

```

Reply:
This is only for `Symbolic`, not for the "normal" `Measurement{T<:AbstractFloat}`, which is what I'd expect for a new generic feature.

Reply:
Aha, right. Sure, let me add that.
I was just surprised that support gap was present and assumed it was unintentional.